src/benchmarklib/listenable.hpp

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <json.hpp>
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace opossum {
+
+enum class Event {
+  ItemRunStarted,
+  ItemRunFinished,
+  CreateReport,
+};
+
+using EventListener = std::function<void(const nlohmann::json& payload)>;
```

Comment:
JSON is a format for serializing something into a file and shouldn’t be abused for passing information around. We could define different message structs and accept a common super class here, but that might only lead to an illusion of type-safety. As the handler should know what type to expect, I’d even consider `std::any`.

Reply:
I will change it to `std::any` then, good idea!